x-pack/platform/plugins/shared/actions/server/application/connector/methods/get_all/get_all.ts

```diff
@@ -81,6 +83,62 @@ export async function getAllUnsecured({
   });
 }
 
+export async function getByIdsWithSecretsDecryptedUnsecured({
+  esClient,
+  encryptedSavedObjectsClient,
+  inMemoryConnectors,
+  spaceId,
+  connectorIds,
+}: GetByIdsWithSecretsUnsecuredParams): Promise<ConnectorWithDecryptedSecrets[]> {
+  const namespace = spaceId && spaceId !== 'default' ? spaceId : undefined;
+  const connectorIdsSet = new Set(connectorIds);
+  const savedObjectsActions: ConnectorWithDecryptedSecrets[] = [];
+  const remainingConnectorIdsToFetch = [...connectorIds];
+  const batchSize = 4; // don't know what value to use here, picked a random one
```

Comment:
I did not find batch variant of getDecryptedAsInternalUser so doing 4 parallel fetches. Batch size I chose is not something I used on purpose, it's a random value as I don't know what it could cost from encryptedSavedObjectsClient point of view.